### PR TITLE
enhance: refine proxy meta cache partition logic

### PR DESCRIPTION
**Pull request by congqixia**
See also #29113

- Unify partition info refresh logic
- Prevent parse partition names for each partition key search request

### CODE REVIEW

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -631,38 +608,75 @@ func (m *MetaCache) showPartitions(ctx context.Context, dbName string, collectio
...
+		name2Info:      name2Info,
+	}
+
+	// Make sure the order of the partition names got every time is the same
+	partitionNames := make([]string, len(infos))
+	for _, info := range infos {
+		partitionName := info.name
+		splits := strings.Split(partitionName, "_")
+		if len(splits) < 2 {
+			log.Debug("partition group not in partitionKey pattern", zap.String("partitionName", partitionName))
```
Comment (yah01) [inline internal/proxy/meta_cache.go:634]: Promote to warning level

File: internal/proxy/meta_cache.go [hunk 2]
```diff
@@ -631,38 +608,75 @@ func (m *MetaCache) showPartitions(ctx context.Context, dbName string, collectio
...
+			return result
+		}
+		index, err := strconv.ParseInt(splits[len(splits)-1], 10, 64)
+		if err != nil {
+			log.Debug("partition group not in partitionKey pattern", zap.String("parititonName", partitionName))
```
Comment (yah01) [inline internal/proxy/meta_cache.go:639]: and miss to print the error

File: internal/proxy/meta_cache.go [hunk 3]
```diff
@@ -631,38 +608,75 @@ func (m *MetaCache) showPartitions(ctx context.Context, dbName string, collectio
...
+	}
+
+	result.indexedPartitionNames = partitionNames
+	return result
+}
+
 func (m *MetaCache) updatePartitions(partitions *milvuspb.ShowPartitionsResponse, database, collectionName string) error {
+	// check partitionID, createdTimestamp and utcstamp has sam element numbers
+	if len(partitions.PartitionNames) != len(partitions.CreatedTimestamps) || len(partitions.PartitionNames) != len(partitions.CreatedUtcTimestamps) {
+		return errors.New("partition names and timestamps number is not aligned, response " + partitions.String())
```
Comment (yah01) [inline internal/proxy/meta_cache.go:652]: merr.WrapErrParameterInvalidMsg

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/proxy/meta_cache.go (hunk 1 above)
Comment (congqixia) [reply internal/proxy/meta_cache.go:634]: No need to be warning during parsing, sync normal partition may not respect partition key naming rule

File: internal/proxy/meta_cache.go (hunk 3 above)
Comment (congqixia) [reply internal/proxy/meta_cache.go:652]: shall be fixed with test case as well

**Pull request merged**